spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc

```diff
@@ -1159,4 +1160,72 @@ Spring Cloud Data Flow will have a suffix appended to it consisting of `-scdf-<t
 schedule name will look like `my-schedule-scdf-tstamp`.
 
-NOTE: The `scdf-` separator is configurable by setting the `spring.cloud.dataflow.task.scheduleNamePrefix` property with the value you prefer.
\ No newline at end of file
+NOTE: The `scdf-` separator is configurable by setting the `spring.cloud.dataflow.task.scheduleNamePrefix` property with the value you prefer.
+
+[[spring-cloud-dataflow-task-cd]]
+== Continuous Deployment
+As task applications evolve, you will want to get your updates to production.  This section will walk through the capabilities that Spring Cloud Data Flow provides around being able to update task applications.
+
+When a task application is registered (<<spring-cloud-dataflow-register-task-apps>>), there is a version associated with it.  A task application can have multiple versions associated with it with one selected as the default.
+
+image::{dataflow-asciidoc}/images/dataflow-task-application-versions.png[Task Application Versions, scaledwidth="50%"]
+
+Versions of an application are managed by registering multiple applications with the same name and coordinates _except_ the version.  For example, if you were to register an application with the following values, you would get one application registered with two versions (2.0.0.RELEASE and 2.1.0.RELEASE):
+
+Application 1
+
+* Name: `timestamp`
+* Type: `task`
+* URI: `maven://org.springframework.cloud.task.app:timestamp-task:2.0.0.RELEASE`
+
+Application 2
+
+* Name: `timestamp`
+* Type: `task`
+* URI: `maven://org.springframework.cloud.task.app:timestamp-task:2.1.0.RELEASE`
+
+Besides having multiple versions, Spring Cloud Data Flow needs to know which version to execute on the next launch.  This is indicated by setting a version to be the default version.  Whatever version of a task application is configured as the default version will be the one to be executed on the next launch request.  You can see which version is the default in the UI as shown below.
+
+image::{dataflow-asciidoc}/images/dataflow-task-default-version.png[Task Application Default Version, scaledwidth="50%"]
+
+=== Task Launch Lifecycle
+In previous versions of Spring Cloud Data Flow, when the request to launch a task was received, Spring Cloud Data Flow would deploy the application (if needed) and execute it.  If the application was being run on a plaform that did not need to have the application deployed every time (CloudFoundry for example), the previously deployed application was used.  This flow has changed in 2.3.  Let's take a look at what happens when a task launch request comes in now.
+
+image::{dataflow-asciidoc}/images/dataflow-task-launch-flow.png[Flow For Launching A Task, scaledwidth="50%"]
+
+There are three main flows to consider in the above diagram.  Launching the first time/launching with no changes is one, the other is launching when there are changes.  We'll look at the flow with no changes first.
```

Comment:
Change "above" to "preceding" (we can't be sure it will be above if someone prints to PDF or some other paging media).

"Launching the first time/launching with no changes is one, the other is launching when there are changes." should be "Launching the first time or launching with no changes is one. The other is launching when there are changes."

"We'll look" should be "We look" (no contraction and present indefinite rather than future tense).